cmd/web.go

```diff
@@ -91,6 +111,9 @@ func runWeb(ctx *cli.Context) error {
 	case setting.HTTP:
 		err = runHTTP(listenAddr, context2.ClearHandler(m))
 	case setting.HTTPS:
+		if setting.RedirectPort80 {
+			go runHTTPRedirector()
```

Comment:
Can you point me at a code example where this sort of thing is being done.  I can't see anything like that happening with the other servers binding to ports in web.go.  They have the same problem that if they stop then the server will stop working.  Presumably in both cases the end users would notice.